metron-platform/metron-parsers/src/main/java/org/apache/metron/parsers/interfaces/MessageParser.java

```diff
@@ -34,15 +34,15 @@
    * @param rawMessage
    * @return If null is returned, this is treated as an empty list.
    */
-  List<T> parse(byte[] rawMessage);
+  List<T> parse(byte[] rawMessage, SensorParserConfig sensorParserConfig);
```

Comment:
Hah, didn't think about Java 8 default methods in interfaces.  Of course, you're right.

Your above last set of changes looks excellent, but in GrokParser the methods configurationUpdate() and init() have to become synchronized methods.  Then I think the code as written allows the configurationUpdated flag to not need further locking, altho different ordering of things might require synchronized get/set/check methods.